### DISCUSSION TITLE
Sensor Trigger When CEC TV Turns On

### DISCUSSION THREAD

**Discussion by asabatino88** [maxileith/homebridge-appletv-enhanced#840] [Q&A] [answered]
Hi - I have an Apple TV that is directly connected to my TV via HDMI. I use the Apple TV Remote (either physical or via iOS) to turn on the Apple TV and the actual TV - I do the same to turn the actual TV off. I'd like to trigger an automation that when the actual TV turns on, my HomeKit enabled TV Backlight turns on.

This seemed simple using this HomeKit plugin; however, it appears that the Apple TV stays on forever (it always shows on in HomeKit, enabled by this plugin). So I can't use the on/off state to trigger my automation. 

I was able to set it up to turn on when content is playing using the "Playing" sensor, but this creates for a wonky experience where when I pause a show the backlight turns off, and then resuming it turns back on.

Is there something I should adjusted on the Apple TV to ensure this plugin views the Apple TV as "off" when I go to power off the TV? I don't want to risk changing the CEC behavior with my TV as it's a pretty flawless existing set up.

**Comment by asabatino88** [comment, marked as answer]
I think the issue might be with my Apple TV having HomePods as the audio source. Is there another clever way of achieving what I’m looking to achieve given this limitation?

> If using audio output devices other than the Apple TV itself, the Apple TV is always reported as powered on. This is a known issue of the dependency [pyatv](https://pyatv.dev/), see https://github.com/postlund/pyatv/issues/1667. As a result, the Apple TV device will only be shown as off in HomeKit when powered off via the Apple TV device in HomeKit. After restarting the plugin the device will always be shown as on.

**Reply by jsiegenthaler** [reply]
Is your TV visible in HomeKit? You could use the TV power state as the trigger. 
What type and model of TV is it?
Does it have an Ethernet port?
Can you use an API call to get the power state?

**Reply by asabatino88** [reply]
Good point! My tv is a Sony Bravia - I could look into getting it into HomeKit and seeing if it shows power state. I’ll explore that route. Thanks for the suggestion.
